recipe-client-addon/lib/EnvExpressions.jsm

```diff
@@ -33,33 +38,53 @@ XPCOMUtils.defineLazyGetter(this, "jexl", () => {
   return jexl;
 });
 
-const getLatestTelemetry = Task.async(function *() {
-  const pings = yield TelemetryArchive.promiseArchivedPingList();
+this.EnvExpressions = {
+  getLatestTelemetry: Task.async(function *() {
+    const pings = yield TelemetryArchive.promiseArchivedPingList();
 
-  // get most recent ping per type
-  const mostRecentPings = {};
-  for (const ping of pings) {
-    if (ping.type in mostRecentPings) {
-      if (mostRecentPings[ping.type].timeStampCreated < ping.timeStampCreated) {
+    // get most recent ping per type
+    const mostRecentPings = {};
+    for (const ping of pings) {
+      if (ping.type in mostRecentPings) {
+        if (mostRecentPings[ping.type].timeStampCreated < ping.timeStampCreated) {
+          mostRecentPings[ping.type] = ping;
+        }
+      } else {
         mostRecentPings[ping.type] = ping;
       }
-    } else {
-      mostRecentPings[ping.type] = ping;
     }
-  }
 
-  const telemetry = {};
-  for (const key in mostRecentPings) {
-    const ping = mostRecentPings[key];
-    telemetry[ping.type] = yield TelemetryArchive.promiseArchivedPingById(ping.id);
-  }
-  return telemetry;
-});
+    const telemetry = {};
+    for (const key in mostRecentPings) {
+      const ping = mostRecentPings[key];
+      telemetry[ping.type] = yield TelemetryArchive.promiseArchivedPingById(ping.id);
+    }
+    return telemetry;
+  }),
+
+  getUserId() {
+    let id = prefs.getCharPref("user_id");
+    console.log("pref user_id", id);
+    if (id === "") {
```

Comment:
> Should we respect a user-set empty value in case privacy add-ons want to get rid of it? That may be a question for outside of this PR.

I doubt there are many privacy sensitive people that will be ok with Normandy as a whole, but not this particular part.